File: fsharp/doublets/test/solver-test.fs
module solver_test

open Expecto
open doublets.solver

[<Tests>]
let tests =
    testList "solver" [
        testCase "with word links found" <| fun _ ->
            Expect.equal
                (doublets "head" "tail") ["head"; "heal"; "teal"; "tell"; "tall"; "tail"] ""
            Expect.equal
                (doublets "door" "lock") ["door"; "boor"; "book"; "look"; "lock"] ""
            Expect.equal
                (doublets "bank" "loan") ["bank"; "bonk"; "book"; "look"; "loon"; "loan"] ""
            Expect.equal
                (doublets "wheat" "bread") ["wheat"; "cheat"; "cheap"; "cheep"; "creep"; "creed"; "breed"; "bread"] ""

        testCase "with no word links found" <| fun _ ->
            Expect.equal
                (doublets "ye" "freezer") [] ""
    ]

[<EntryPoint>]
let main argv =
    runTestsInAssembly defaultConfig argv

File: csharp/doublets/test/solver-test.cs
namespace doublets;

using Xunit;
using static doublets.solver;

public class solver_test {
    [Fact]
    public void with_word_links_found() {
        Assert.Equal (
            new string[] { "head", "heal", "teal", "tell", "tall", "tail" },
            doublets("head", "tail")
        );
        Assert.Equal (
            new string[] { "door", "boor", "book", "look", "lock" },
            doublets("door", "lock")
        );
        Assert.Equal (
            new string[] { "bank", "bonk", "book", "look", "loon", "loan" },
            doublets("bank", "loan")
        );
        Assert.Equal (
            new string[] { "wheat", "cheat", "cheap", "cheep", "creep", "creed", "breed", "bread" },
            doublets("wheat", "bread")
        );
    }

    [Fact]
    public void with_no_word_links_found() {
        Assert.Equal (
            new string [] {},
            doublets("ye", "freezer")
        );
    }
}
